api/compaction_manager.cc

```diff
@@ -117,6 +121,23 @@ void set_compaction_manager(http_context& ctx, routes& r) {
         });
     });
 
+    cm::stop_keyspace_compaction.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        auto ks_name = validate_keyspace(ctx, req->param);
+        auto table_names = split_cf(req->get_query_param("tables"));
+        if (table_names.empty()) {
+            table_names = map_keys(ctx.db.local().find_keyspace(ks_name).metadata().get()->cf_meta_data());
+        }
+        auto type = req->get_query_param("type");
+        co_await ctx.db.invoke_on_all([&ks_name, &table_names, type] (database& db) {
+            auto& cm = db.get_compaction_manager();
+            return parallel_for_each(table_names, [&db, &cm, &ks_name, type] (sstring& table_name) {
+                auto& t = db.find_column_family(ks_name, table_name);
```

Comment:
I'm not familiar enough with this area of the code, so I have to ask: if find_column_family() or validate_keyspace() above throw - will this provide the user with a decent error message or a mess?

Reply:
By the way, thanks to this review I found a bug in Alternator, which also used this find_column_family in its UpdateTable request, so results in a funky exception when given a non-existent table instead of the appropriate error message :-)

Reply:
If you're curious - https://github.com/scylladb/scylla/issues/9747

Reply:
> I'm not familiar enough with this area of the code, so I have to ask: if find_column_family() or validate_keyspace() above throw - will this provide the user with a decent error message or a mess?

```
$ ./scylla-cli.py compaction_manager stop_keyspace_compaction/{keyspace} POST --keyspace foo --type CLEANUP
{'message': 'Keyspace foo Does not exist', 'code': 400}

$ ./scylla-cli.py compaction_manager stop_keyspace_compaction/{keyspace} POST --keyspace system_schema --tables foo,bar --type CLEANUP
{'message': "std::_Nested_exception<no_such_column_family> (Can't find a column family foo in keyspace system_schema): std::out_of_range ()", 'code': 500}

Not a total mess but the table-not-found error is thrown as a nested exception that isn't very elegant, like what you encountered in #9747

Reply:
Exactly what I worried about. Note that's not just the textual error message - it's also a different error code (400 vs 500). In Alternator the different error code is important, because 400 is a real error in the request, while 500 (internal server error) means the user should retry the request - and the DynamoDB driver indeed retries the same request multiple times.
I don't know how much of a difference it makes in our REST API because it isn't documented anywhere :-(